src/helix/impl/props.cljc

```diff
@@ -126,7 +126,7 @@
   [style]
   (cond
     ;; React Native allows arrays of styles
-    (vector? style) (into-js-array (map primitive-obj style))
+    (vector? style) (into-js-array (map #(if (map? %) (primitive-obj %) %) style))
     ;; when map, convert to an object w/ camel casing
     (map? style) (primitive-obj style)
```

Comment:
can we swap the order of these while we're at it? I think it's more likely we'll hit the `map?` branch than the `vector?` branch typically.